### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by cruwaller**
* Code clean ups
* `-Wall`, `-Werror` and `-Wextra` enabled to get trigger code flaws during the compile time
* Linker time optimizations (`-flto`) enabled to make code size smaller
* STM8 build fixed. Not functional but builds :)

### CODE REVIEW

File: src/src/rtc6705.h [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 void rtc6705spiPinSetup(void);
 void rtc6705writeRegister(uint32_t data);
-uint32_t rtc6705readRegister(uint8_t register);
```
Comment (cruwaller) [inline src/src/rtc6705.h:27]: `register` is reserved keyword :)

File: src/src/targets/Eachine_TX526/Eachine_TX526.c [hunk 2]
```diff
@@ -3,10 +3,8 @@
 #include "helpers.h"
 #include <Arduino.h>
 
-struct PowerMapping power_mapping[] = {
-  {0, 0}, {25, 14}, {200, 23},
-};
-uint8_t power_mapping_size = ARRAY_SIZE(power_mapping);
+
+uint8_t saPowerLevelsLut[SA_NUM_POWER_LEVELS] = {RACE_MODE, 1, 14, 23};
```
Comment (JyeSmith) [inline src/src/targets/Eachine_TX526/Eachine_TX526.c:7]: "1" should be first in the array for Low power on Disarm.

File: src/src/targets/Eachine_TX801/Eachine_TX801.c [hunk 3]
```diff
@@ -3,10 +3,8 @@
 #include "helpers.h"
 #include <Arduino.h>
 
-struct PowerMapping power_mapping[4] = {
-  {0, 0}, {25, 14}, {100, 20}, {200, 23},
-};
-uint8_t power_mapping_size = ARRAY_SIZE(power_mapping);
+
+uint8_t saPowerLevelsLut[SA_NUM_POWER_LEVELS] = {RACE_MODE, 1, 14, 20, 23};
```
Comment (JyeSmith) [inline src/src/targets/Eachine_TX801/Eachine_TX801.c:7]: "1" should be first in the array for Low power on Disarm.

**Review by JyeSmith — CHANGES_REQUESTED** [review]

File: src/src/targets/Eachine_TX526/Eachine_TX526.c (hunk 2 above)
Comment (cruwaller) [reply src/src/targets/Eachine_TX526/Eachine_TX526.c]: fixed.

File: src/src/targets/Eachine_TX801/Eachine_TX801.c (hunk 3 above)
Comment (cruwaller) [reply src/src/targets/Eachine_TX801/Eachine_TX801.c]: fixed.

**Comment by cruwaller** [comment]
Don't merge this yet. #33 to merged before this and a bit more testing is needed.

**Review by JyeSmith — APPROVED** [review]

**Pull request merged**